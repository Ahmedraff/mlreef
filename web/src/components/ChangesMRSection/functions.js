import { Base64 } from 'js-base64';
import Base64ToArrayBuffer from 'base64-arraybuffer';
import { getFileDifferences } from '../../functions/apiCalls';
import CommitsApi from 'apis/CommitsApi';
import { isImageFormat } from 'functions/dataParserHelpers';

const commitsApi = new CommitsApi();

export const fetchFileDiffInfo = (projectId, aheadCommits) => Promise.all(aheadCommits.map(async (commit, index) => {
  const { id: commitId } = commit;
  const commitDiffs = await commitsApi.getCommitDiff(projectId, commitId, 0, false);
  return await Promise.all(commitDiffs?.body?.map(async (commitDiff) => {
    const {
      previousVersionFile,
      nextVersionFile,
      imageFileSize,
    } = await getFileDifferences(projectId, commitDiff, commit.parent_ids[0], commit.id);
    let previousVersionFileParsed = previousVersionFile;
    let nextVersionFileParsed = nextVersionFile;
    let sizeDeleted = 0;
    let sizeAdded = 0;
    const fileName = commitDiff.old_path.split('/').slice(-1)[0];
    if (previousVersionFile && !isImageFormat(fileName)) {
      previousVersionFileParsed = Base64.decode(
        Base64ToArrayBuffer.encode(previousVersionFile),
      );
      sizeDeleted = previousVersionFile.byteLength;
    }
    if (nextVersionFile && !isImageFormat(fileName)) {
      nextVersionFileParsed = Base64.decode(Base64ToArrayBuffer.encode(nextVersionFile));
      sizeAdded = nextVersionFile.byteLength;
    }
    const path = commitDiff.new_path
      ? commitDiff.new_path
      : commitDiff.old_path;
    const fileExtension = path.split('.').pop().toLowerCase();
    return {
      fileName,
      fileExtension,
      previousVersionFileParsed,
      nextVersionFileParsed,
      id: `${commit.id} ${index}`,
      sizeDeleted,
      sizeAdded,
      imageFileSize,
    };
  }));
}));
