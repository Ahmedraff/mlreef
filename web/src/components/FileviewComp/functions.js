import FilesApi from 'apis/FilesApi.ts';
import CommitsApi from 'apis/CommitsApi.ts';

const commitsApi = new CommitsApi();

const filesApi = new FilesApi();

const getCommit = (projectId, lastCommitId) => commitsApi
  .getCommitDetails(projectId, lastCommitId);

const getFileAndInformation = (gid, file, branch, commit) => filesApi
  .getFileData(gid, file?.includes('/') ? encodeURIComponent(file) : file, branch || commit)
  .then(async (fData) => {
    const commitInfoDet = await getCommit(gid, fData.last_commit_id);
    return {
      fData,
      commitInfoDet,
    };
  });

export default {
  getFileAndInformation,
};
