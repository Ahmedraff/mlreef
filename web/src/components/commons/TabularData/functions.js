import genUUID from 'uuid/v1';

const parseNumber = (text) => {
  const num = parseFloat(text);

  return Number.isNaN(num) ? text : num;
};

export const transpose = (matrix) => matrix[0].map((_, ci) => matrix.map((row) => row[ci]));

const sanatizeMatrix = (matrix) => {
  const head = matrix[0];
  const tail = matrix[matrix.length - 1];

  const sameLength = head.length === tail.length;
  const lastRowfilled = tail.every((col) => col !== undefined);

  return sameLength && lastRowfilled ? matrix : matrix.slice(0, -1);
};

export const cvsToArray = (text, separator = ',') => {
  const matrix = text.split('\n')
    .map((line) => line.split(separator)
      .map(parseNumber));

  return sanatizeMatrix(matrix);
};

export const jsonToArray = (input) => {
  const items = Array.isArray(input) ? input : Object.values(input);

  if (!items.length) return items;
  // const row = Object.values(input)[0];
  const row = items[0];
  const header = Object.keys(row).map((label) => label);

  // const data = Object.values(input)
  const data = items
    .map((cols) => Object.values(cols).map((val) => val));

  const matrix = [header].concat(data);

  return sanatizeMatrix(matrix);
};

export const arrayToRichData = (matrix, separator = ',') => {
  const header = matrix[0];

  return {
    id: genUUID(),
    meta: {
      separator,
      timestamp: new Date(),
    },
    matrix,
    // data: flat(matrix.map((row, y) => row.map((value, x) => ({ x, y, value }))))
    data: matrix.map((cols, y) => ({
      id: y,
      cols: cols.map((value, x) => ({
        x,
        y,
        value,
        label: header[x],
      })),
    })),
  };
};
