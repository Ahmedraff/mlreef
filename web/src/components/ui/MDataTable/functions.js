const extractField = (row, x) => row.cols.find((col) => col.x === x)?.value;

export const sortDesc = (x) => (rows) => rows.slice()
  .sort((a, b) => extractField(b, x) - extractField(a, x));

export const sortAsc = (x) => (rows) => rows.slice()
  .sort((a, b) => extractField(a, x) - extractField(b, x));

export const round = (decs) => (input) => {
  const decimals = parseInt(decs, 10);

  if (Number.isNaN(decimals)) return input;

  if (decimals > 12) return input;

  const num = parseFloat(input);
  if (Number.isNaN(num)) return input;

  return Math.round(num * (10 ** decimals)) / (10 ** decimals);
};
