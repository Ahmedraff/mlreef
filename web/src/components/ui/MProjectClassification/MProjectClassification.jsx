import React, { Component } from 'react';
import { bool, shape, arrayOf, string } from 'prop-types';
import { CircularProgress } from '@material-ui/core';
import { Link } from 'react-router-dom';
import ArrowButton from 'components/arrow-button/arrowButton';
import ProjectSet from '../../projectSet';
import './MProjectClassification.scss';
import MCheckBox from '../MCheckBox/MCheckBox';

class MProjectClassification extends Component {
  projFilterBtnsList = ['own', 'starred', 'explore'];

  constructor(props) {
    super(props);
    this.state = {
      isDataTypesVisible: true,
    };
  }

  // this change tabs in projectSet
  changeScreen = (screen) => {
    const {
      history: {
        push,
        location: {
          pathname,
        },
      },
    } = this.props;

    push(`${pathname}${screen}`);
  }

  handleClickDataTypesButton = () => this.setState(
    (prevState) => ({
      isDataTypesVisible: !prevState.isDataTypesVisible,
    }),
  );

  handleProjectFilterBtn(e, screen) {
    this.changeScreen(screen);
    const { classification } = this.props;
    this.projFilterBtnsList.forEach((btnId) => {
      document.getElementById(`${classification}-${btnId}-btn`).classList.replace('btn-basic-info', 'btn-basic-dark');
    });
    e.target.classList.replace('btn-basic-dark', 'btn-basic-info');
  }

  render() {
    const {
      isDataTypesVisible,
    } = this.state;

    const {
      classification,
      isFetching,
      userProjects,
      starredProjects,
      allProjects,
      history: {
        location: {
          hash: screen,
        },
      },
    } = this.props;
    const dataTypes = [
      { name: `${classification} data-types`, label: 'Text' },
      { name: `${classification} data-types`, label: 'Image' },
      { name: `${classification} data-types`, label: 'Audio' },
      { name: `${classification} data-types`, label: 'Video' },
      { name: `${classification} data-types`, label: 'Tabular' },
    ];

    return (
      <div className="main-content">
        <div id="buttons-div">
          <div id="filter-div">
            <button
              id={`${classification}-own-btn`}
              key={this.ownBtn}
              onClick={(e) => this.handleProjectFilterBtn(e, '#personal')}
              type="button"
              className="btn btn-basic-dark"
            >
              Own
            </button>
            <button
              id={`${classification}-starred-btn`}
              key={this.starredBtn}
              onClick={(e) => this.handleProjectFilterBtn(e, '#starred')}
              type="button"
              className="btn btn-basic-dark"
            >
              Starred
            </button>
            <button
              id={`${classification}-explore-btn`}
              key={this.exploreBtn}
              onClick={(e) => this.handleProjectFilterBtn(e, '#explore')}
              type="button"
              className="btn btn-basic-dark"
            >
              Explore
            </button>
          </div>
          <div id="new-element-container">
            <Link
              to="/new-project"
              type="button"
              className="btn btn-primary"
            >
              New Project
            </Link>
          </div>
        </div>
        {isFetching
          ? (
            <div className="project-content-loader">
              <CircularProgress size={40} />
            </div>
          )
          : (
            <div id="main-content-div">
              <ProjectSet
                screen={screen || '#personal'}
                changeScreen={this.changeScreen}
                allProjects={allProjects}
                personalProjects={userProjects}
                starredProjects={starredProjects}
              />
              <div id="side-filters">
                <div id="input-div">
                  <p>Refine by</p>
                </div>
                <br />
                <div id="data-types-deploy-btn">
                  <p>
                    Data types
                  </p>
                  <ArrowButton callback={this.handleClickDataTypesButton} />
                </div>
                {isDataTypesVisible
                && (
                  dataTypes.map((dtype) => (
                    <MCheckBox
                      key={`${dtype.name} ${dtype.label} comp`}
                      name={dtype.name}
                      labelValue={dtype.label}
                      callback={(name, labelValue, newValue) => {
                        
                      }}
                    />
                  ))
                )}
              </div>
            </div>
          )}
      </div>
    );
  }
}

MProjectClassification.propTypes = {
  classification: string.isRequired,
  isFetching: bool.isRequired,
  userProjects: arrayOf(shape({})),
  starredProjects: arrayOf(shape({})),
  allProjects: arrayOf(shape({})),
};

MProjectClassification.defaultProps = {
  userProjects: [],
  starredProjects: [],
  allProjects: [],
};

export default MProjectClassification;
