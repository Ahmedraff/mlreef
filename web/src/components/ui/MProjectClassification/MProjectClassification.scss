#ml-project, #model, #data-operation, #data-visualization {
  max-width: unset;
}

#no-proj-found-div {
  width: 100%;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  p {
    color: var(--secondary);   
  }
}