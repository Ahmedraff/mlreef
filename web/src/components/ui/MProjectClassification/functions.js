import { ALGORITHM, DATA_OPERATION, MODEL, OPERATION, VISUALIZATION } from "dataTypes";

export const getSearchable = (feSearchableTypes) => {
  switch (feSearchableTypes) {
    case DATA_OPERATION:
      return OPERATION;
    case MODEL:
      return ALGORITHM;
    default:
      return VISUALIZATION;
  }
}