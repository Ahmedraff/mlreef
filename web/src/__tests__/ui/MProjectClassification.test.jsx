import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { ML_PROJECT } from 'dataTypes';
import MProjectClassification from 'components/ui/MProjectClassification/MProjectClassification';
import { projectsArrayMock } from 'testData';
import { storeFactory } from 'functions/testUtils';
import { parseToCamelCase } from 'functions/dataParserHelpers';
import ArrowButton from 'components/arrow-button/arrowButton';

const setup = () => shallow(
  <MProjectClassification
    classification={ML_PROJECT}
    userProjects={[]}
    starredProjects={[]}
    allProjects={projectsArrayMock.projects.all.map((p) => parseToCamelCase(p))}
    history={{ push: () => {}, location: { hash: '#Personal'}}}
  />,
);

global.ResizeObserver = () => ({ observe: jest.fn() });

test('test html elements', () => {
  const tree = renderer.create(
    <Provider store={storeFactory()}>
      <MemoryRouter>
        <MProjectClassification
          classification={ML_PROJECT}
          userProjects={[]}
          starredProjects={[]}
          allProjects={projectsArrayMock.projects.all.map((p) => parseToCamelCase(p))}
          history={{ push: () => {}, location: { hash: '#Personal' }}}
        />
      </MemoryRouter>
    </Provider>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('test functionality', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });
  test('assert that exploring filter buttons change styles when clicked and receive the right params', () => {
    const personalBtn = wrapper.find(`#${ML_PROJECT}-personal-btn`);
    const starredBtn = wrapper.find(`#${ML_PROJECT}-starred-btn`);
    const exploreBtn = wrapper.find(`#${ML_PROJECT}-explore-btn`);
    const mockedFunc = jest.fn();
    const darkClass = 'btn-basic-dark';
    wrapper.instance().handleProjectFilterBtn = mockedFunc;
    personalBtn.simulate('click', {});
    expect(mockedFunc).toHaveBeenCalledWith({}, '#personal');
    expect(personalBtn.hasClass(darkClass)).toBe(true);

    starredBtn.simulate('click', {});
    expect(mockedFunc).toHaveBeenCalledWith({}, '#starred');
    expect(personalBtn.hasClass(darkClass)).toBe(true);

    exploreBtn.simulate('click', {});
    expect(mockedFunc).toHaveBeenCalledWith({}, '#explore');
    expect(exploreBtn.hasClass(darkClass)).toBe(true);
  });

  test('assert that side filter collapse buttons work', () => {
    const arrBtnDataTypes = wrapper.find('ArrowButton').at(0);
    arrBtnDataTypes.dive().find('button').simulate('click', {});
    expect(wrapper.state().isDataTypesVisible).toBe(false);
    const arrBtnFramework = wrapper.find(ArrowButton).at(1);
    arrBtnFramework.dive().find('button').simulate('click', {});
    expect(wrapper.state().isFrameworksVisible).toBe(false);
    const arrBtnModelType = wrapper.find(ArrowButton).at(2);
    arrBtnModelType.dive().find('button').simulate('click', {});
    expect(wrapper.state().isModelTypesVisible).toBe(false);
    const arrBtnMlCat = wrapper.find(ArrowButton).at(3);
    arrBtnMlCat.dive().find('button').simulate('click', {});
    expect(wrapper.state().isMlCategoriesVisible).toBe(false);
  });
});
