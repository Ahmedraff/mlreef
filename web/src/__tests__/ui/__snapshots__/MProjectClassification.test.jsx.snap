// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test html elements 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-around",
    }
  }
>
  <div>
    <div
      id="buttons-div"
    >
      <div
        id="filter-div"
      >
        <button
          className="btn btn-basic-dark"
          id="ml-project-personal-btn"
          onClick={[Function]}
          type="button"
        >
          My projects
        </button>
        <button
          className="btn btn-basic-dark"
          id="ml-project-starred-btn"
          onClick={[Function]}
          type="button"
        >
          Starred
        </button>
        <button
          className="btn btn-basic-dark"
          id="ml-project-explore-btn"
          onClick={[Function]}
          type="button"
        >
          Explore
        </button>
      </div>
      <div
        className="ml-auto"
        id="new-element-container"
      >
        <a
          className="btn btn-primary"
          href="/new-project/classification/ml-project"
          onClick={[Function]}
          type="button"
        >
          New project
        </a>
      </div>
    </div>
    <div
      className="m-project-classification"
    >
      <div
        id="cards-section"
      >
        <div
          className="card"
        >
          <div
            className="card-container project-card-container"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            <p
              className="card-title"
            />
            <div
              className="project-card-container-autor d-flex "
            >
              <p>
                MLReef
              </p>
            </div>
            <div
              className="card-content"
            >
              <div
                className="d-flex"
              >
                <div
                  className="mr-2"
                >
                  <i
                    className="fa fa-file t-success"
                  >
                    <span
                      className="label"
                    >
                       Text
                    </span>
                  </i>
                </div>
                <div
                  className="mr-2"
                >
                  <i
                    className="fa fa-volume-up t-info"
                  >
                    <span
                      className="label"
                    >
                       Audio
                    </span>
                  </i>
                </div>
                <div
                  className=""
                >
                  <i
                    className="fa fa-video t-danger"
                  >
                    <span
                      className="label"
                    >
                       Video
                    </span>
                  </i>
                </div>
                <div
                  className=""
                >
                  <i
                    className="fas fa-grip-lines-vertical t-warning"
                  >
                    <span
                      className="label"
                    >
                       Tabular
                    </span>
                  </i>
                </div>
              </div>
              <p>
                Categorical Classification of alphabets in the American Sign Language
              </p>
              <div
                className="d-flex t-secondary"
              >
                <div
                  className="mr-3"
                >
                  <i
                    className="fa fa-star"
                  >
                    <span
                      className="label"
                    >
                       0
                    </span>
                  </i>
                </div>
                <div
                  className="mr-3"
                >
                  <i
                    className="fa fa-code-branch"
                  >
                    <span
                      className="label"
                    >
                       0
                    </span>
                  </i>
                </div>
              </div>
            </div>
            <div
              className="card-actions"
            >
              <div
                className=""
                style={
                  Object {
                    "display": "flex",
                    "marginRight": "-1rem",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
      <div
        id="side-filters"
      >
        <div
          id="input-div"
        >
          <p>
            Refine by
          </p>
          <button>
            Clear filters
          </button>
        </div>
        <br />
        <div
          className="name-filter-section"
        >
          <p>
            Data types
          </p>
          <div
            className="dropdown-btn-container-div"
          >
            <button
              aria-label="dropdown-btn"
              className="btn btn-icon btn-hidden fa fa-chevron-down p-1"
              id=""
              onClick={[Function]}
              type="button"
            />
            
          </div>
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project dataTypes Text"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Text
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project dataTypes Image"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Image
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project dataTypes Audio"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Audio
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project dataTypes Video"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Video
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project dataTypes Tabular"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Tabular
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="name-filter-section"
        >
          <p>
            Framework
          </p>
          <div
            className="dropdown-btn-container-div"
          >
            <button
              aria-label="dropdown-btn"
              className="btn btn-icon btn-hidden fa fa-chevron-down p-1"
              id=""
              onClick={[Function]}
              type="button"
            />
            
          </div>
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project framework TensorFlow"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            TensorFlow
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project framework Pytorch"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Pytorch
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project framework Keras"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Keras
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project framework Scikit Learn"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Scikit Learn
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="name-filter-section"
        >
          <p>
            Model Type
          </p>
          <div
            className="dropdown-btn-container-div"
          >
            <button
              aria-label="dropdown-btn"
              className="btn btn-icon btn-hidden fa fa-chevron-down p-1"
              id=""
              onClick={[Function]}
              type="button"
            />
            
          </div>
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project modelTypes CNN"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            CNN
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project modelTypes Clustering"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Clustering
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project modelTypes Trees"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Trees
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project modelTypes Regression"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Regression
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="name-filter-section"
        >
          <p>
            ML categories
          </p>
          <div
            className="dropdown-btn-container-div"
          >
            <button
              aria-label="dropdown-btn"
              className="btn btn-icon btn-hidden fa fa-chevron-down p-1"
              id=""
              onClick={[Function]}
              type="button"
            />
            
          </div>
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project mlCategories Regression"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Regression
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project mlCategories Prediction"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Prediction
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project mlCategories Classification"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Classification
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
        <div
          className="m-checkbox "
          htmlFor="ml-project mlCategories Dimensionality reduction"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <p>
            Dimensionality reduction
          </p>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
